src/components/TextInput/index.tsx

```diff
@@ -1,28 +1,32 @@
 import React, {useEffect, useRef} from 'react';
-import _ from 'underscore';
+import type {StyleProp, ViewStyle} from 'react-native';
 import * as Browser from '@libs/Browser';
 import DomUtils from '@libs/DomUtils';
 import Visibility from '@libs/Visibility';
 import useThemeStyles from '@styles/useThemeStyles';
 import BaseTextInput from './BaseTextInput';
-import * as baseTextInputPropTypes from './BaseTextInput/baseTextInputPropTypes';
+import type BaseTextInputProps from './BaseTextInput/types';
+import type {BaseTextInputRef} from './BaseTextInput/types';
 import * as styleConst from './styleConst';
 
-function TextInput(props) {
+type RemoveVisibilityListener = () => void;
+
+function TextInput(props: BaseTextInputProps, ref: BaseTextInputRef) {
     const styles = useThemeStyles();
-    const textInputRef = useRef(null);
-    const removeVisibilityListenerRef = useRef(null);
+    const textInputRef = useRef<HTMLElement | null>(null);
+    const removeVisibilityListenerRef = useRef<RemoveVisibilityListener>(null);
 
     useEffect(() => {
+        let removeVisibilityListener = removeVisibilityListenerRef.current;
         if (props.disableKeyboard) {
-            textInputRef.current.setAttribute('inputmode', 'none');
+            textInputRef.current?.setAttribute('inputmode', 'none');
         }
 
         if (props.name) {
-            textInputRef.current.setAttribute('name', props.name);
+            textInputRef.current?.setAttribute('name', props.name);
         }
 
-        removeVisibilityListenerRef.current = Visibility.onVisibilityChange(() => {
+        removeVisibilityListener = Visibility.onVisibilityChange(() => {
```

Comment:
Why this logic was updated?

Reply:
@VickyStash logic is not changed, this ref is used only in this use effect on the mount I am creating new variable in this effect and assigning the value to it and then when its unmount its calling this function which will be assigned , did that because earlier typescript was complaining because of `Cannot assign to 'current' because it is a read-only property.` and with that which I did we are surpassing this error

Reply:
@kubabutkiewicz alternatively we can turn it back to as it was and have it like
`(removeVisibilityListenerRef.current as RemoveVisibilityListener) = Visibility.onVisibilityChange(() => {`

Reply:
When its like that its still giving eslint errors. I will leave it as it is
<img width="1074" alt="image" src="https://github.com/Expensify/App/assets/25230417/286519fb-374e-4e07-9d04-4ef8df0219af">
<img width="1239" alt="image" src="https://github.com/Expensify/App/assets/25230417/62d388ab-e138-4771-8d7e-40b1a4312967">